### ISSUE TITLE
Hydro Thunder - Time records music bug

### ISSUE THREAD

**Issue by PixelCrusader** [libretro/flycast#171]
Windows 7 64 bit, AMD Radeon HD 6450, Intel i5-2320 CPU @ 3.00GHz
RetroArch 1.7.3 - Reicast 0.1 db11a17

The game is running too fast. Also, the music that plays on the records screen after entering initials continues to play after exiting to the track menu.

Game speed references: [1](https://www.youtube.com/watch?v=Tuedmc9Hgjo) [2](https://www.youtube.com/watch?v=rk7PUWLo_po) [3](https://www.youtube.com/watch?v=4W1SvOp_K8A) 

Record music references: [1](https://www.youtube.com/watch?v=Tuedmc9Hgjo#t=3m15s) [2](https://www.youtube.com/watch?v=rk7PUWLo_po#t=3m27s) [3](https://www.youtube.com/watch?v=lPdByQEMRHE&t=667s#t=18m28s) [4](https://www.youtube.com/watch?v=6rtBRRUyccE#t=4m42s)

**Comment by PixelCrusader** [comment]
Windows 7 64 bit, AMD Radeon HD 6450, Intel i5-2320 CPU @ 3.00GHz
RetroArch 1.7.4 - Reicast 0.1 917d3d9

The game is now running at the proper speed. Maybe related to #245?

**Comment by PixelCrusader** [comment]
Windows 7 64 bit, AMD Radeon HD 6450, Intel i5-2320 CPU @ 3.00GHz
RetroArch 1.7.4 2a7a2e7dee - Reicast 0.1 514c569

The game is now unplayable. Loading the game with threaded rendering turned on will take you though the startup sequence before staying on a black screen, and launching with threaded rendering turned off will take you through the startup screen and freeze RetroArch.

Attempts to roll the Reicast core back to 917d3d9 didn't change the outcome.  Possibly related to RetroArch 2a7a2e7dee and #260

**Comment by PixelCrusader** [comment]
Windows 7 64 bit, AMD Radeon HD 6450, Intel i5-2320 CPU @ 3.00GHz
RetroArch 1.7.5 0e93fade3e Reicast 01ffff9

I've managed to get the game working again.

It looks like the big problem was that the game was set to run with the cable type set to TV(RGB). It's been too long since I first had this problem, so I cannot say with certainty if it was a matter of TV(RGB) becoming unusable for the game on September 12th, or if it was never usuable and updating changed a setting, or purely an error I made. I treated TV(RGB) as the standard choice for running games in Reicast, but it looks like Hydro Thunder and Street Fighter Alpha 3 don't like that very much.

Launching the game in TV(RGB) will lead to the problems I already described above: the game not proceeding past the start-up screen if Threaded Rendering is enabled and causing RetroArch to crash if Threaded Rendering is turned off. Using VGA(RGB) will launch the game, albeit without the Sega Enterprises screen in the start up.

I also discovered that the game's UI will glitch out, quickly switching between normal and being stretched downward if Threaded Rendering is enabled. It felt like this was more likely to happen if you went to the save and load screen, although this might just be coincidental:

![hydro thunder usa rerelease -181128-045212](https://user-images.githubusercontent.com/40935002/49203365-fa3fab80-f375-11e8-858c-ac512a6a9afc.png)

On the upside, the game now fits the display area much better by default. I used to have to make a cfg file with a much more modified custom aspect ratio and X/Y pos to make the game look right. Now at default the game only vertically off by a few pixels, creating a line of discoloration at the top of the screen at the start and at the end of races:

![hydro thunder usa rerelease -181129-014223](https://user-images.githubusercontent.com/40935002/49204021-720ed580-f378-11e8-9075-d56d143aed76.png)

I should also add that YMMV based on the version of the game you are using. My version is the .gdi later release version. Another .cdi version I tried did not respond to my settings changes and was still unplayable.

**Comment by flyinghead** [comment]
Which region of the game are you using?
The PAL version doesn't support VGA and will not load with either TV modes. However the NTSC-US version seems to work fine with any setting.

**Comment by PixelCrusader** [comment]
Should be the US version, both of them.

Okay, now it looks like I -really- figured it out.

I removed retroarch-core-options.cfg to start from scratch and launched Hydro Thunder again. It now works, as well as the .cdi version I mentioned. The Sega Enterprises screen displays as well.

I compared my old settings to the defaults. It's the Broadcast option. The game will run with TV(RGB) but only if the Broadcast option is set to NTSC or PAL_M. My Broadcast setting was on Default, which led to the problems outlined above.

I guess using Default causes the emulator to not be able to choose between the two possible Broadcast modes, thus preventing the game from launching. That's just me swinging at the dark, though. I'm just an end user.

If you set the Cable Type to VGA(RGB) you can play the game with any Broadcast setting you want. For this game VGA(RGB) does a much better job of filling the screen without Custom Aspect Ratios than TV(RGB) does, making it prefererable for windowed play. However, turning on Threaded Rendering while using VGA(RGB) will cause the game's menu UI and starting credits to flicker between being normal and being stretched downward.

All of that was in regard to the .gdi US rerelease version. The .cdi version refuses to run unless Broadcast is set to NTSC or PAL_M, regardless of Cable setting. It also shows the same aspect ratio regardless of Cable option, the .gdi TV(RGB) aspect ratio. This means the threaded rendering UI problems don't occur. (The .cdi also has a Sega Dreamcast logo at the bottom of the title screen that the .gdi doesn't for some reason.)

(It also looks like the .gdi US version of Rush 2049 is similarly sensitive to this. Any Broadcast setting besides NTSC and PAL_M will send you to the bios menu, regardles of Cable option.)

Long story short: Set Broadcast to NTSC to run it, Set Cable to VGA(RGB) if you want the best windowed or non aspect ratio customized experience. Set to TV(RGB) and modify with Custom Aspect Ratio and X/Y Pos if you want Threaded Rendering. .cdi behaves like TV(RGB) regardless of Cable setting.

All that accounted for, the only problem the game has is the records music bug.

**Comment by flyinghead** [comment]
The "Default" broadcast setting is in fact a hack. It basically means "whatever the game wants". It works in most cases but not for this game. Similar problem with San Francisco Rush 2049.
The same applies to the "Default" region setting, which is a similar hack.

In case of problem it's always best to specify the correct broadcast/region for the game.

**Comment by flyinghead** [comment]
The music on the ranking screen should now stop when going back to the main menu.

**Comment by PixelCrusader** [comment]
RetroArch 1.8.4, Flycast (0.1 5dd73cc)

Confirming that the ranking screen music is fixed. Outside of having to resize the screen a bit if you can't stand black borders at the top and bottom , it looks like the game is running without any problems. I'll close the issue.

Thanks!
